logstash-core/src/test/java/org/logstash/common/io/DeadLetterQueueReaderTest.java

```diff
@@ -543,6 +543,23 @@ public void testSeekByTimestampWhenAllSegmentsAreDeleted() throws IOException, I
         }
     }
 
+    @Test
+    public void testSeekByTimestampWhenSegmentIs1Byte() throws IOException, InterruptedException {
+        final long startTime = System.currentTimeMillis();
+        com.google.common.io.Files.touch(dir.resolve("1.log").toFile());
```

Comment:
I think it should insert one byte in the offending segment, the [version byte](https://github.com/elastic/logstash/blob/46babd6041eba476ba61abd5c3ce5486dc7f2997/logstash-core/src/main/java/org/logstash/common/io/RecordIOWriter.java#L99), to be adherent to reality